algorithms.py

```diff
@@ -15,58 +17,138 @@
 X = np.random.normal(size = (m, n))
 X = X - np.mean(X, axis=1, keepdims=True)
 
+W1 = np.random.normal(size = (k, m))
+W2 = np.random.normal(size = (m, k))
+
+# auxiliary function that computes the distance between two SVDs
+def compute_dist(W2,k,u_svd):
```

Comment:
for readability, we prefer space after commas, except possibly when indexing in brackets